src/main.ts

```diff
@@ -1,24 +1,80 @@
-import './style.css'
-import typescriptLogo from './typescript.svg'
-import viteLogo from '/vite.svg'
-import { setupCounter } from './counter.ts'
+import { AuthComponent } from "./components/AuthComponent";
+import { FeedComponent } from "./components/FeedComponent";
+import { AuthService } from "./services/AuthService";
 
-document.querySelector<HTMLDivElement>('#app')!.innerHTML = `
-  <div>
-    <a href="https://vitejs.dev" target="_blank">
-      <img src="${viteLogo}" class="logo" alt="Vite logo" />
-    </a>
-    <a href="https://www.typescriptlang.org/" target="_blank">
-      <img src="${typescriptLogo}" class="logo vanilla" alt="TypeScript logo" />
-    </a>
-    <h1>Vite + TypeScript</h1>
-    <div class="card">
-      <button id="counter" type="button"></button>
-    </div>
-    <p class="read-the-docs">
-      Click on the Vite and TypeScript logos to learn more
-    </p>
-  </div>
-`
+// POPULATE LOCAL STORAGE SCRIPT
+function populateLocalStorage(): void {
+  const users = [
+    { username: "ramiro", password: "password", email: "ramiro@example.com" },
+    { username: "maria", password: "password", email: "maria@example.com" },
+  ];
 
-setupCounter(document.querySelector<HTMLButtonElement>('#counter')!)
+  const posts = [
+    {
+      username: "ramiro",
+      content: "here is my cat!",
+      image: "http://placekitten.com/200/300",
+      timestamp: 1705335002353,
+    },
+    {
+      username: "ramiro",
+      content: "hello there!",
+      image: "",
+      timestamp: 1705335002352,
+    },
+    {
+      username: "maria",
+      content: "hello world",
+      image: "",
+      timestamp: 1705335002351,
+    },
+  ];
+
+  localStorage.setItem("users", JSON.stringify(users));
+  localStorage.setItem("posts", JSON.stringify(posts));
+}
+
+populateLocalStorage();
+
+// EVENT LISTENERS
```

Comment:
you can try to follow the behavioral pattern or event delegation patter here, review this doc: https://javascript.info/event-delegation